zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -383,16 +383,59 @@ pub(crate) fn get_transparent_balances<P: consensus::Parameters>(
 pub(crate) fn add_transparent_account_balances(
     conn: &rusqlite::Connection,
     mempool_height: BlockHeight,
+    min_confirmations: u32,
     account_balances: &mut HashMap<AccountId, AccountBalance>,
 ) -> Result<(), SqliteClientError> {
-    let mut stmt_account_balances = conn.prepare(
+    // TODO (#1592): Ability to distinguish between Transparent pending change and pending non-change
+    let mut stmt_account_spendable_balances = conn.prepare(
         "SELECT u.account_id, SUM(u.value_zat)
          FROM transparent_received_outputs u
          JOIN transactions t
          ON t.id_tx = u.transaction_id
-         -- the transaction that created the output is mined or is definitely unexpired
+         -- the transaction that created the output is mined and with enough confirmations
          WHERE (
             t.mined_height < :mempool_height -- tx is mined
+            AND :mempool_height - t.mined_height >= :min_confirmations -- has at least min_confirmations
```

Comment:
This looks wrong to me. `mempool_height` is typically `chain_tip_height + 1`. `mined_height` is at most the chain tip height, or null, so changing the query in this fashion is introducing a footgun for zero-conf shielding.

Also, `:mempool_height - t.mined_height` will be `NULL` when `mined_height IS NULL`, which then will always cause this condition to fail.

Instead of using `:mempool_height` directly, compute a `balance_height` outside of the query from the mempool height and `min_confirmations`, and then query against that height here, accounting for `mined_height` being allowed to be null in the case that `min_confirmations == 0`.

Reply:
> This looks wrong to me. `mempool_height` is typically `chain_tip_height + 1`. `mined_height` is at most the chain tip height, or null, so changing the query in this fashion is introducing a footgun for zero-conf shielding.

Is Zero-conf shielding something that should be address here? I understood it was not. 

> Also, `:mempool_height - t.mined_height` will be `NULL` when `mined_height IS NULL`, which then will always cause this condition to fail.

In this particular query a pre-condition is that the TX is mined so I guess desirable that the condition is evaluated to `FALSE`. 

> 
> Instead of using `:mempool_height` directly, compute a `balance_height` outside of the query from the mempool height and `min_confirmations`, and then query against that height here, accounting for `mined_height` being allowed to be null in the case that `min_confirmations == 0`.
This sounds like a good idea.

Reply:
> Is Zero-conf shielding something that should be address here? I understood it was not.

No, we'll need another PR for that. The current code appears to work as documented: `min_confirmations == 0` behaves the same as `min_confirmations == 1`. I don't think we should complicate this PR to make zero-conf shielding work; it's obvious that this code will need to change.

Reply:
Btw @nuttycom did you see this? https://github.com/zcash/librustzcash/issues/1413#issuecomment-2439717322